Application crash when fast clicking the folders inside the file dialog

### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When I fast click the folders inside the file selection widget, the app crashes

### How to reproduce

- Open Fyne Demo
- Select Dialogs
- Select any of the file dialogs (file Open, Folder Open)
- Start clicking very fast the (Parent) Icon / area.
- Tha app crashes
```
alex@cosmo:/dev/shm/fyne/cmd/fyne_demo$ go run main.go 
2023/09/18 11:02:49 Lifecycle: Started
2023/09/18 11:02:49 Lifecycle: Entered Foreground
2023/09/18 11:03:02 Cancelled
2023/09/18 11:03:12 Cancelled
fatal error: sync: RUnlock of unlocked RWMutex

goroutine 16 [running, locked to thread]:
sync.fatal({0xc83529?, 0xc00024f500?})
	/usr/local/go/src/runtime/panic.go:1061 +0x18
sync.(*RWMutex).rUnlockSlow(0xc0000ff8c0, 0x29b098?)
	/usr/local/go/src/sync/rwmutex.go:129 +0x39
sync.(*RWMutex).RUnlock(0xc00236f560?)
	/usr/local/go/src/sync/rwmutex.go:119 +0x25
panic({0xc14700?, 0xc0015041c8?})
	/usr/local/go/src/runtime/panic.go:914 +0x21f
fyne.io/fyne/v2/dialog.(*fileDialog).setView.func3(0x438c32?, {0xdacbe0?, 0xc00024f490?})
	/dev/shm/fyne/dialog/file.go:489 +0x113
fyne.io/fyne/v2/widget.(*gridWrapLayout).setupGridItem(0xc001e17d00, 0xc00024f500, 0x0, 0x60?)
	/dev/shm/fyne/widget/gridwrap.go:528 +0xc6
fyne.io/fyne/v2/widget.(*gridWrapLayout).updateGrid(0xc001e17d00, 0x1)
	/dev/shm/fyne/widget/gridwrap.go:615 +0x5ff
fyne.io/fyne/v2/widget.(*gridWrapLayout).Layout(0x8?, {0xc0023d9410?, 0x6d81cd?, 0xc0039ba8c8?}, {0x42a00000?, 0x42b80000?})
	/dev/shm/fyne/widget/gridwrap.go:480 +0x18
fyne.io/fyne/v2.(*Container).layout(...)
	/dev/shm/fyne/container.go:195
fyne.io/fyne/v2.(*Container).Resize(0xc260c0?, {0xda3330?, 0x0?})
	/dev/shm/fyne/container.go:168 +0x57
fyne.io/fyne/v2/internal/widget.(*scrollContainerRenderer).Layout(0xc001bb8cc0, {0x43a80000, 0x43500000})
	/dev/shm/fyne/internal/widget/scroller.go:276 +0x8a
fyne.io/fyne/v2/internal/widget.(*Base).Resize(0xc00023f490, {0xda6670?, 0x0?})
	/dev/shm/fyne/internal/widget/base.go:59 +0x13b
fyne.io/fyne/v2/internal/widget.(*Scroll).Resize(0xc00023f490, {0xda6670?, 0x0?})
	/dev/shm/fyne/internal/widget/scroller.go:452 +0x36
fyne.io/fyne/v2/widget.(*gridWrapRenderer).Layout(0xdad648?, {0x23d2e70?, 0xc0?})
	/dev/shm/fyne/widget/gridwrap.go:333 +0x17
fyne.io/fyne/v2/widget.(*BaseWidget).Resize(0xc0023d2e70, {0xc68a57?, 0x0?})
	/dev/shm/fyne/widget/widget.go:61 +0x134
fyne.io/fyne/v2/widget.(*GridWrap).Resize(0xc0023d2e70, {0x0?, 0x0?})
	/dev/shm/fyne/widget/gridwrap.go:148 +0x1e
fyne.io/fyne/v2/layout.(*paddedLayout).Layout(0x8?, {0xc001265680, 0x1, 0xc0039ba8c0?}, {0x42b00000?, 0x42c80000?})
	/dev/shm/fyne/layout/paddedlayout.go:21 +0x8d
fyne.io/fyne/v2.(*Container).layout(...)
	/dev/shm/fyne/container.go:195
fyne.io/fyne/v2.(*Container).Resize(0xc260c0?, {0xda3330?, 0x0?})
	/dev/shm/fyne/container.go:168 +0x57
fyne.io/fyne/v2/internal/widget.(*scrollContainerRenderer).Layout(0xc001bb8a20, {0x43ac0000, 0x43580000})
	/dev/shm/fyne/internal/widget/scroller.go:276 +0x8a
fyne.io/fyne/v2/internal/widget.(*Base).Resize(0xc00023f2d0, {0x42000000?, 0x42000000?})
	/dev/shm/fyne/internal/widget/base.go:59 +0x13b
fyne.io/fyne/v2/internal/widget.(*Scroll).Resize(0xc00023f2d0, {0x269cb58?, 0xc0?})
	/dev/shm/fyne/internal/widget/scroller.go:452 +0x36
fyne.io/fyne/v2/layout.(*borderLayout).Layout(0xc001e17f00, {0xc001e17ec0, 0x3, 0xc0023d9798?}, {0x23d9788?, 0xc0?})
	/dev/shm/fyne/layout/borderlayout.go:61 +0x4f3
fyne.io/fyne/v2/internal/driver/common.(*Canvas).updateLayout(0xc0000ff8c0, {0xdab078?, 0xc002464820})
	/dev/shm/fyne/internal/driver/common/canvas.go:603 +0xd1
fyne.io/fyne/v2/internal/driver/common.(*Canvas).EnsureMinSize.func1(0xc002f6e4c0, {0x4365a200?, 0x42d00000?})
	/dev/shm/fyne/internal/driver/common/canvas.go:110 +0xfc
fyne.io/fyne/v2/internal/driver/common.(*Canvas).walkTree.func2({0xdab2b8?, 0xc00023f340?}, {0x2464800?, 0xc0?}, {0xdab078?, 0xc002464820?})
	/dev/shm/fyne/internal/driver/common/canvas.go:520 +0xda
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xdab078, 0xc002464820}, 0x0, {0xdab198, 0xc00023f810}, {0x43e60000?, 0x44200000?}, {0x2639c20?, 0xc0?}, {0xdab378?, ...}, ...)
	/dev/shm/fyne/internal/driver/util.go:199 +0x46d
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
	/dev/shm/fyne/internal/driver/util.go:178
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xdab198, 0xc00023f810}, 0x0, {0xdab078, 0xc0024649b0}, {0xc5a540?, 0x0?}, {0x2639c08?, 0xc0?}, {0xf8475800?, ...}, ...)
	/dev/shm/fyne/internal/driver/util.go:192 +0x41f
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
	/dev/shm/fyne/internal/driver/util.go:178
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xdab078, 0xc0024649b0}, 0x0, {0xdac5e0, 0xc00264b780}, {0x23d9b60?, 0xc0?}, {0xa66425?, 0x0?}, {0xc65fb3?, ...}, ...)
	/dev/shm/fyne/internal/driver/util.go:192 +0x41f
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
	/dev/shm/fyne/internal/driver/util.go:178
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xdac5e0, 0xc00264b780}, 0x0, {0x0, 0x0}, {0x23d9bf0?, 0xc0?}, {0xa79525?, 0x0?}, {0x23d9bd8?, ...}, ...)
	/dev/shm/fyne/internal/driver/util.go:192 +0x41f
fyne.io/fyne/v2/internal/driver.WalkVisibleObjectTree({0xdac5e0?, 0xc00264b780?}, 0xc000340460?, 0xa79500?)
	/dev/shm/fyne/internal/driver/util.go:134 +0x46
fyne.io/fyne/v2/internal/driver/common.(*Canvas).walkTree(0xa602c2?, 0xc00236f560, 0x0, 0xc0023d9d70)
	/dev/shm/fyne/internal/driver/common/canvas.go:526 +0x15f
fyne.io/fyne/v2/internal/driver/common.(*Canvas).WalkTrees(0xc0000ff8c0, 0xc0023d9d68?, 0x42db05?)
	/dev/shm/fyne/internal/driver/common/canvas.go:447 +0xd9
fyne.io/fyne/v2/internal/driver/common.(*Canvas).EnsureMinSize(0xc0000ff8c0)
	/dev/shm/fyne/internal/driver/common/canvas.go:143 +0x150
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow.func1()
	/dev/shm/fyne/internal/driver/glfw/loop.go:189 +0x2b
fyne.io/fyne/v2/internal/driver/glfw.(*window).RunWithContext(0xc000341d50?, 0xc0023d9e30)
	/dev/shm/fyne/internal/driver/glfw/window.go:922 +0x43
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow(0x4?, 0x0?)
	/dev/shm/fyne/internal/driver/glfw/loop.go:188 +0x45
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).drawSingleFrame(0xc0023d9f90?)
	/dev/shm/fyne/internal/driver/glfw/loop.go:95 +0x1a5
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).startDrawThread.func1()
	/dev/shm/fyne/internal/driver/glfw/loop.go:244 +0x1d5
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).startDrawThread in goroutine 1
	/dev/shm/fyne/internal/driver/glfw/loop.go:220 +0xbb

goroutine 1 [select, locked to thread]:
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).runGL(0xc000341d50)
	/dev/shm/fyne/internal/driver/glfw/loop.go:115 +0x15c
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run(0xc000341d50)
	/dev/shm/fyne/internal/driver/glfw/driver.go:168 +0x6f
fyne.io/fyne/v2/internal/driver/glfw.(*window).ShowAndRun(0xc0003ce3c0)
	/dev/shm/fyne/internal/driver/glfw/window.go:231 +0x26
main.main()
	/dev/shm/fyne/cmd/fyne_demo/main.go:68 +0x4cd

goroutine 6 [chan receive]:
fyne.io/fyne/v2/test.NewApp.func1()
	/dev/shm/fyne/test/testapp.go:166 +0x45
created by fyne.io/fyne/v2/test.NewApp in goroutine 1
	/dev/shm/fyne/test/testapp.go:164 +0x325

goroutine 7 [IO wait]:
internal/poll.runtime_pollWait(0x7f95146a9b30, 0x72)
	/usr/local/go/src/runtime/netpoll.go:343 +0x85
internal/poll.(*pollDesc).wait(0xc00008a4e0?, 0xc00040fed0?, 0x1)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc00008a4e0, {0xc00040fed0, 0x10000, 0x10000})
	/usr/local/go/src/internal/poll/fd_unix.go:164 +0x27a
os.(*File).read(...)
	/usr/local/go/src/os/file_posix.go:29
os.(*File).Read(0xc000062140, {0xc00040fed0?, 0x0?, 0x0?})
	/usr/local/go/src/os/file.go:118 +0x52
github.com/fsnotify/fsnotify.(*Watcher).readEvents(0xc000094af0)
	/home/alex/go/pkg/mod/github.com/fsnotify/fsnotify@v1.6.0/backend_inotify.go:356 +0xd2
created by github.com/fsnotify/fsnotify.NewWatcher in goroutine 1
	/home/alex/go/pkg/mod/github.com/fsnotify/fsnotify@v1.6.0/backend_inotify.go:150 +0x186

goroutine 8 [chan receive]:
fyne.io/fyne/v2/app.watchFile.func1()
	/dev/shm/fyne/app/settings_desktop.go:43 +0x66
created by fyne.io/fyne/v2/app.watchFile in goroutine 1
	/dev/shm/fyne/app/settings_desktop.go:42 +0xf8

goroutine 10 [chan receive]:
github.com/godbus/dbus/v5.newConn.func1()
	/home/alex/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/conn.go:303 +0x2c
created by github.com/godbus/dbus/v5.newConn in goroutine 1
	/home/alex/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/conn.go:302 +0x4f6

goroutine 18 [IO wait]:
internal/poll.runtime_pollWait(0x7f95146a9a38, 0x72)
	/usr/local/go/src/runtime/netpoll.go:343 +0x85
internal/poll.(*pollDesc).wait(0xc0000c3880?, 0xc0012803d0?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).ReadMsg(0xc0000c3880, {0xc0012803d0, 0x10, 0x10}, {0xc0003b0620, 0x1000, 0x1000}, 0x14?)
	/usr/local/go/src/internal/poll/fd_unix.go:301 +0x38a
net.(*netFD).readMsg(0xc0000c3880, {0xc0012803d0?, 0xc001380600?, 0x1490498?}, {0xc0003b0620?, 0xc001280170?, 0xc001280172?}, 0xc00128016c?)
	/usr/local/go/src/net/fd_posix.go:78 +0x31
net.(*UnixConn).readMsg(0xc000062158, {0xc0012803d0?, 0xc001380600?, 0xda12e0?}, {0xc0003b0620?, 0xdaa470?, 0x1bcbe60?})
	/usr/local/go/src/net/unixsock_posix.go:115 +0x45
net.(*UnixConn).ReadMsgUnix(0xc000062158, {0xc0012803d0?, 0x0?, 0x0?}, {0xc0003b0620?, 0x436845?, 0xc0012a4510?})
	/usr/local/go/src/net/unixsock.go:143 +0x36
github.com/godbus/dbus/v5.(*oobReader).Read(0xc0003b0600, {0xc0012803d0?, 0xc0001a1e00?, 0x436bc5?})
	/home/alex/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/transport_unix.go:21 +0x3c
io.ReadAtLeast({0xda17e0, 0xc0003b0600}, {0xc0012803d0, 0x10, 0x10}, 0x10)
	/usr/local/go/src/io/io.go:335 +0x90
io.ReadFull(...)
	/usr/local/go/src/io/io.go:354
github.com/godbus/dbus/v5.(*unixTransport).ReadMessage(0xc0000133e0)
	/home/alex/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/transport_unix.go:91 +0xfe
github.com/godbus/dbus/v5.(*Conn).inWorker(0xc0001fd4a0)
	/home/alex/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/conn.go:390 +0x43
created by github.com/godbus/dbus/v5.(*Conn).Auth in goroutine 1
	/home/alex/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/auth.go:118 +0x945

goroutine 67 [chan receive]:
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).catchTerm(0x0?)
	/dev/shm/fyne/internal/driver/glfw/driver_desktop.go:172 +0x85
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run in goroutine 1
	/dev/shm/fyne/internal/driver/glfw/driver.go:167 +0x65

goroutine 12 [IO wait]:
internal/poll.runtime_pollWait(0x7f95146a9940, 0x72)
	/usr/local/go/src/runtime/netpoll.go:343 +0x85
internal/poll.(*pollDesc).wait(0xc00008a6c0?, 0xc00050fed0?, 0x1)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc00008a6c0, {0xc00050fed0, 0x10000, 0x10000})
	/usr/local/go/src/internal/poll/fd_unix.go:164 +0x27a
os.(*File).read(...)
	/usr/local/go/src/os/file_posix.go:29
os.(*File).Read(0xc000062168, {0xc00050fed0?, 0x0?, 0x0?})
	/usr/local/go/src/os/file.go:118 +0x52
github.com/fsnotify/fsnotify.(*Watcher).readEvents(0xc000094cd0)
	/home/alex/go/pkg/mod/github.com/fsnotify/fsnotify@v1.6.0/backend_inotify.go:356 +0xd2
created by github.com/fsnotify/fsnotify.NewWatcher in goroutine 1
	/home/alex/go/pkg/mod/github.com/fsnotify/fsnotify@v1.6.0/backend_inotify.go:150 +0x186

goroutine 13 [chan receive]:
fyne.io/fyne/v2/app.watchFile.func1()
	/dev/shm/fyne/app/settings_desktop.go:43 +0x66
created by fyne.io/fyne/v2/app.watchFile in goroutine 1
	/dev/shm/fyne/app/settings_desktop.go:42 +0xf8

goroutine 14 [chan receive]:
fyne.io/fyne/v2/app.watchFreedekstopThemeChange()
	/dev/shm/fyne/app/app_xdg.go:194 +0x1c5
created by fyne.io/fyne/v2/app.watchTheme in goroutine 1
	/dev/shm/fyne/app/app_xdg.go:166 +0x1a

goroutine 34 [select]:
fyne.io/fyne/v2/internal/async.(*UnboundedFuncChan).processing(0xc0003971d0)
	/dev/shm/fyne/internal/async/chan_func.go:45 +0xd5
created by fyne.io/fyne/v2/internal/async.NewUnboundedFuncChan in goroutine 1
	/dev/shm/fyne/internal/async/chan_func.go:22 +0xe5

goroutine 35 [chan receive]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/dev/shm/fyne/internal/driver/common/window.go:34 +0x47
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/dev/shm/fyne/internal/driver/glfw/window.go:959 +0x139

goroutine 36 [chan receive]:
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).refreshSystrayMenu.func1()
	/dev/shm/fyne/internal/driver/glfw/driver_desktop.go:139 +0x34
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).refreshSystrayMenu in goroutine 1
	/dev/shm/fyne/internal/driver/glfw/driver_desktop.go:138 +0xee

goroutine 37 [select]:
fyne.io/fyne/v2/internal/async.(*UnboundedFuncChan).processing(0xc0004869c0)
	/dev/shm/fyne/internal/async/chan_func.go:45 +0xd5
created by fyne.io/fyne/v2/internal/async.NewUnboundedFuncChan in goroutine 1
	/dev/shm/fyne/internal/async/chan_func.go:22 +0xe5

goroutine 38 [runnable]:
syscall.Syscall(0x48b2f9?, 0x476aa9?, 0x80?, 0xbcc8c0?)
	/usr/local/go/src/syscall/syscall_linux.go:69 +0x25
syscall.Close(0x4ae965?)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:320 +0x25
internal/poll.(*SysFile).destroy(...)
	/usr/local/go/src/internal/poll/fd_unixjs.go:24
internal/poll.(*FD).destroy(0xc002650240)
	/usr/local/go/src/internal/poll/fd_unix.go:81 +0x51
internal/poll.(*FD).decref(0xc000390690?)
	/usr/local/go/src/internal/poll/fd_mutex.go:213 +0x53
internal/poll.(*FD).Close(0xc002650240)
	/usr/local/go/src/internal/poll/fd_unix.go:104 +0x45
os.(*file).close(0xc002650240)
	/usr/local/go/src/os/file_unix.go:315 +0x98
os.(*File).Close(0xc002370000?)
	/usr/local/go/src/os/file_posix.go:23 +0x1b
os.ReadDir({0xd9e01c?, 0xc0031ce000?})
	/usr/local/go/src/os/dir.go:124 +0x18d
fyne.io/fyne/v2/internal/repository.(*FileRepository).List(0x7f9514672e60?, {0x7f9514672e60?, 0xc001265d00?})
	/dev/shm/fyne/internal/repository/file.go:218 +0x45
fyne.io/fyne/v2/storage.List({0x7f9514672e60, 0xc001265d00})
	/dev/shm/fyne/storage/uri.go:479 +0x4f
fyne.io/fyne/v2/storage.(*legacyListable).List(0x9b4289?)
	/dev/shm/fyne/storage/file.go:47 +0x1d
fyne.io/fyne/v2/dialog.(*fileDialog).refreshDir(0xc001bd3880, {0xdaeb20, 0xc0014f8a80})
	/dev/shm/fyne/dialog/file.go:339 +0x76
fyne.io/fyne/v2/dialog.(*fileDialog).setLocation(0xc001bd3880, {0x7f9514672e60, 0xc001265d00})
	/dev/shm/fyne/dialog/file.go:452 +0x716
fyne.io/fyne/v2/dialog.(*fileDialog).makeUI.func7(0x9f7666?)
	/dev/shm/fyne/dialog/file.go:225 +0x39
fyne.io/fyne/v2/widget.(*List).Select.func1()
	/dev/shm/fyne/widget/list.go:212 +0x57
fyne.io/fyne/v2/widget.(*List).Select(0xc000ccd5c0, 0x1)
	/dev/shm/fyne/widget/list.go:217 +0x184
fyne.io/fyne/v2/dialog.(*fileDialog).setLocation(0xc001bd3880, {0xdab1f8, 0xc0023b0ff0})
	/dev/shm/fyne/dialog/file.go:394 +0x1c6
fyne.io/fyne/v2/dialog.(*fileDialog).setSelected(0xc001bd3880, {0xdab1f8, 0xc0023b0ff0}, 0x0)
	/dev/shm/fyne/dialog/file.go:460 +0x170
fyne.io/fyne/v2/dialog.(*fileDialog).setView.func4(0x9f7666?)
	/dev/shm/fyne/dialog/file.go:496 +0x45
fyne.io/fyne/v2/widget.(*GridWrap).Select.func1()
	/dev/shm/fyne/widget/gridwrap.go:172 +0x57
fyne.io/fyne/v2/widget.(*GridWrap).Select(0xc0023d2e70, 0x0)
	/dev/shm/fyne/widget/gridwrap.go:177 +0x184
fyne.io/fyne/v2/widget.(*gridWrapLayout).setupGridItem.func1()
	/dev/shm/fyne/widget/gridwrap.go:538 +0x96
fyne.io/fyne/v2/widget.(*gridWrapItem).Tapped(0xc00024f500, 0xc001aa4901)
	/dev/shm/fyne/widget/gridwrap.go:421 +0x33
fyne.io/fyne/v2/internal/driver/glfw.(*window).mouseClickedHandleTapDoubleTap.func1()
	/dev/shm/fyne/internal/driver/glfw/window.go:649 +0x22
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/dev/shm/fyne/internal/driver/common/window.go:35 +0x38
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/dev/shm/fyne/internal/driver/glfw/window.go:959 +0x139

goroutine 69 [syscall]:
os/signal.signal_recv()
	/usr/local/go/src/runtime/sigqueue.go:152 +0x29
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:23 +0x13
created by os/signal.Notify.func1.1 in goroutine 67
	/usr/local/go/src/os/signal/signal.go:151 +0x1f

goroutine 52 [chan receive]:
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).refreshSystrayMenu.func1()
	/dev/shm/fyne/internal/driver/glfw/driver_desktop.go:139 +0x34
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).refreshSystrayMenu in goroutine 15
	/dev/shm/fyne/internal/driver/glfw/driver_desktop.go:138 +0xee

goroutine 53 [chan receive]:
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).refreshSystrayMenu.func1()
	/dev/shm/fyne/internal/driver/glfw/driver_desktop.go:139 +0x34
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).refreshSystrayMenu in goroutine 15
	/dev/shm/fyne/internal/driver/glfw/driver_desktop.go:138 +0xee

goroutine 40 [select]:
fyne.io/systray.stayRegistered()
	/home/alex/go/pkg/mod/fyne.io/systray@v1.10.1-0.20230722100817-88df1e0ffa9a/systray_unix.go:266 +0x2af
created by fyne.io/systray.nativeStart in goroutine 1
	/home/alex/go/pkg/mod/fyne.io/systray@v1.10.1-0.20230722100817-88df1e0ffa9a/systray_unix.go:230 +0xbbc
exit status 2
```

### Screenshots

https://github.com/fyne-io/fyne/assets/11015965/78de73a4-37d4-4f50-826e-52d4541887f9



### Example code

No code. Replicated with Fyne Demo

### Fyne version

v2.4.0

### Go compiler version

1.21.1

### Operating system and version

Ubuntu 18.04

### Additional Information

_No response_

I still got a crash in the file dialog in my app while I was doing some legit file searching
``
fatal error: sync: RUnlock of unlocked RWMutex

goroutine 25 [running, locked to thread]:
sync.fatal({0xce79bf?, 0xcb81c0?})
        /usr/local/go/src/runtime/panic.go:1061 +0x18
sync.(*RWMutex).rUnlockSlow(0xc0002330e0, 0x48b519?)
        /usr/local/go/src/sync/rwmutex.go:129 +0x39
sync.(*RWMutex).RUnlock(0xc0015e8a00?)
        /usr/local/go/src/sync/rwmutex.go:119 +0x25
panic({0xc72820?, 0xc00072dce0?})
        /usr/local/go/src/runtime/panic.go:914 +0x21f
fyne.io/fyne/v2/widget.ellipsisPriorBound.func1(0x8, 0x10)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/richtext.go:862 +0xbb
fyne.io/fyne/v2/widget.binarySearch(0xc0038a2890, 0x8, 0x10)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/richtext.go:826 +0x34
fyne.io/fyne/v2/widget.ellipsisPriorBound({0xc002b47490, 0x2, 0x2}, 0x8?, 0x42800000, 0xc0038a2ed8)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/richtext.go:865 +0x1a7
fyne.io/fyne/v2/widget.lineBounds(0xc0015a0b40, 0x2, 0x2, 0x42800000, {0xc0?, 0x495acc?}, 0xc0038a2ed8)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/richtext.go:934 +0x19bb
fyne.io/fyne/v2/widget.(*RichText).updateRowBounds.func1({0xc0014bc630?, 0x1, 0x0?})
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/richtext.go:410 +0x20e
fyne.io/fyne/v2/widget.(*RichText).updateRowBounds(0xc000ebe780)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/richtext.go:457 +0x1b8
fyne.io/fyne/v2/widget.(*RichText).Refresh(0xc000ebe780)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/richtext.go:100 +0x2b
fyne.io/fyne/v2/widget.(*Label).Refresh(0xc00148ed00)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/label.go:100 +0x2d
fyne.io/fyne/v2/dialog.(*fileItemRenderer).Layout(0xc001598980, {0x1a2f50?, 0xc0?})
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/dialog/fileitem.go:105 +0x19b
fyne.io/fyne/v2/internal/driver/common.(*Canvas).updateLayout(0xc0002330e0, {0xe19b58?, 0xc0001a2f50})
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/common/canvas.go:609 +0x188
fyne.io/fyne/v2/internal/driver/common.(*Canvas).EnsureMinSize.func1(0xc000f22080, {0x432b4400?, 0x43aa2800?})
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/common/canvas.go:110 +0xfc
fyne.io/fyne/v2/internal/driver/common.(*Canvas).walkTree.func2({0xe19138?, 0xc00148ed00?}, {0x1a2f00?, 0xc0?}, {0xe19b58?, 0xc0001a2f50?})
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/common/canvas.go:520 +0xda
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xe19b58, 0xc0001a2f50}, 0x0, {0xe19fd8, 0xc0001a2fc0}, {0x1?, 0x0?}, {0x0?, 0x0?}, {0x3c0b6b0?, ...}, ...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:199 +0x46d
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:178
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xe19fd8, 0xc0001a2fc0}, 0x0, {0xe190d8, 0xc0013e0c80}, {0xcb81c0?, 0x0?}, {0x2173338?, 0xc0?}, {0xf8475800?, ...}, ...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:192 +0x41f
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:178
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xe190d8, 0xc0013e0c80}, 0x0, {0xe19978, 0xc00036a8c0}, {0xcb81c0?, 0x0?}, {0x2173320?, 0xc0?}, {0xf8475800?, ...}, ...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:192 +0x41f
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:178
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xe19978, 0xc00036a8c0}, 0x0, {0xe1a038, 0xc001b22000}, {0x40a850?, 0xc0?}, {0x38a3688?, 0xc0?}, {0x7c5518?, ...}, ...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:192 +0x41f
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:178
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xe1a038, 0xc001b22000}, 0x0, {0xe190d8, 0xc0013e0aa0}, {0xcb81c0?, 0x0?}, {0x2173308?, 0xc0?}, {0xf8475800?, ...}, ...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:192 +0x41f
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:178
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xe190d8, 0xc0013e0aa0}, 0x0, {0xe19978, 0xc00036a1c0}, {0x43d90000?, 0x4431c000?}, {0x2172f30?, 0xc0?}, {0xe19918?, ...}, ...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:192 +0x41f
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:178
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xe19978, 0xc00036a1c0}, 0x0, {0xe190d8, 0xc0013e12c0}, {0xcb81c0?, 0x0?}, {0x162a870?, 0xc0?}, {0xf8475800?, ...}, ...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:192 +0x41f
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:178
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xe190d8, 0xc0013e12c0}, 0x0, {0xe198b8, 0xc00036ac40}, {0x43d90000?, 0x4431c000?}, {0x162a720?, 0xc0?}, {0xe19618?, ...}, ...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:192 +0x41f
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:178
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xe198b8, 0xc00036ac40}, 0x0, {0xe190d8, 0xc0013e14f0}, {0xcb81c0?, 0x0?}, {0x162a708?, 0xc0?}, {0xf8475800?, ...}, ...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:192 +0x41f
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:178
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xe190d8, 0xc0013e14f0}, 0x0, {0xe19df8, 0xc001a94980}, {0xcca436?, 0x0?}, {0x0?, 0x0?}, {0x38a3b70?, ...}, ...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:192 +0x41f
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:178
fyne.io/fyne/v2/internal/driver.walkObjectTree({0xe19df8, 0xc001a94980}, 0x0, {0x0, 0x0}, {0x38a3bf0?, 0xc0?}, {0xae7425?, 0x0?}, {0x23811f8?, ...}, ...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:192 +0x41f
fyne.io/fyne/v2/internal/driver.WalkVisibleObjectTree({0xe19df8?, 0xc001a94980?}, 0xc000036220?, 0xcca436?)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/util.go:134 +0x46
fyne.io/fyne/v2/internal/driver/common.(*Canvas).walkTree(0xaced82?, 0xc0015e8a00, 0x0, 0xc0038a3d70)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/common/canvas.go:526 +0x15f
fyne.io/fyne/v2/internal/driver/common.(*Canvas).WalkTrees(0xc0002330e0, 0xc0038a3d68?, 0x42db05?)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/common/canvas.go:447 +0xd9
fyne.io/fyne/v2/internal/driver/common.(*Canvas).EnsureMinSize(0xc0002330e0)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/common/canvas.go:143 +0x150
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow.func1()
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/loop.go:192 +0x2b
fyne.io/fyne/v2/internal/driver/glfw.(*window).RunWithContext(0xc000164000?, 0xc0038a3e30)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/window.go:924 +0x43
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow(0x4?, 0x0?)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/loop.go:191 +0x45
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).drawSingleFrame(0xc0038a3f90?)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/loop.go:88 +0x1a5
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).startDrawThread.func1()
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/loop.go:245 +0x1c5
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).startDrawThread in goroutine 1
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/loop.go:223 +0xbb

goroutine 1 [sync.RWMutex.Lock, locked to thread]:
sync.runtime_SemacquireRWMutex(0x4258a00042a00000?, 0x20?, 0xc000ebe7a8?)
        /usr/local/go/src/runtime/sema.go:87 +0x25
sync.(*RWMutex).Lock(0xc0014bc630?)
        /usr/local/go/src/sync/rwmutex.go:152 +0x6a
fyne.io/fyne/v2/widget.(*RichText).updateRowBounds(0xc000ebe780)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/richtext.go:460 +0x1f2
fyne.io/fyne/v2/widget.(*RichText).Refresh(0xc000ebe780)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/richtext.go:100 +0x2b
fyne.io/fyne/v2/widget.(*Label).Refresh(0xc00148ed00)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/label.go:100 +0x2d
fyne.io/fyne/v2/widget.(*Label).SetText(...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/label.go:118
fyne.io/fyne/v2/dialog.(*fileItemRenderer).Refresh(0xc001598980)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/dialog/fileitem.go:120 +0x6c
fyne.io/fyne/v2/widget.(*BaseWidget).Refresh(0xc001fd8501?)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/widget.go:140 +0x26
fyne.io/fyne/v2/dialog.(*fileDialogItem).setLocation(0xc0001a2f50, {0x7f43c924a580?, 0xc003c0b8e0?}, 0x80?, 0x0)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/dialog/fileitem.go:55 +0xe5
fyne.io/fyne/v2/dialog.(*fileDialog).setView.func3(0xc, {0xe19b58?, 0xc0001a2f50})
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/dialog/file.go:536 +0xed
fyne.io/fyne/v2/widget.(*gridWrapLayout).setupGridItem(0xc0020c7240, 0xc0001a2fc0, 0xc, 0x0?)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/gridwrap.go:528 +0xc6
fyne.io/fyne/v2/widget.(*gridWrapLayout).updateGrid(0xc0020c7240, 0x0)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/gridwrap.go:618 +0x5ff
fyne.io/fyne/v2/widget.(*gridWrapLayout).offsetUpdated(...)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/widget/gridwrap.go:508
fyne.io/fyne/v2/internal/widget.(*Scroll).updateOffset(0xc00036a8c0, 0x0, 0xc1c80000)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/widget/scroller.go:488 +0x1ee
fyne.io/fyne/v2/internal/widget.(*Scroll).scrollBy(0xc00036a8c0, 0x0, 0xc1c80000)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/widget/scroller.go:469 +0xc5
fyne.io/fyne/v2/internal/widget.(*Scroll).Scrolled(0xa9150f?, 0xe19978?)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/widget/scroller.go:462 +0x1d
fyne.io/fyne/v2/internal/driver/glfw.(*window).processMouseScrolled(0xc0004ea000, 0x0, 0xbff0000000000000)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/window.go:701 +0x203
fyne.io/fyne/v2/internal/driver/glfw.(*window).mouseScrolled(0xae1925?, 0xa94fc0?, 0x0?, 0xbff0000000000000?)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/window_desktop.go:396 +0x99
github.com/go-gl/glfw/v3.3/glfw.goScrollCB(0x1a?, 0x0, 0xbff0000000000000)
        /home/alex/go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20221017161538-93cebf72946b/input.go:353 +0x3e
github.com/go-gl/glfw/v3.3/glfw._Cfunc_glfwPollEvents()
        _cgo_gotypes.go:1595 +0x3f
github.com/go-gl/glfw/v3.3/glfw.PollEvents()
        /home/alex/go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20221017161538-93cebf72946b/window.go:964 +0xf
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).tryPollEvents(0xc0026f9b50?)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/loop_desktop.go:34 +0x30
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).runGL(0xc000607b98?)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/loop.go:118 +0x197
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run(0xc000164000)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/driver.go:168 +0x6f
fyne.io/fyne/v2/internal/driver/glfw.(*window).ShowAndRun(0xc0004ea000)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/window.go:227 +0x26
github.com/alexballas/go2tv/internal/gui.Start({0xe15388?, 0xc0003aa4c0}, 0xc0004ea1e0)
        /home/alex/test/go2tv/internal/gui/gui.go:126 +0x4fa
main.run()
        /home/alex/test/go2tv/cmd/go2tv/go2tv.go:110 +0xd5a
main.main()
        /home/alex/test/go2tv/cmd/go2tv/go2tv.go:51 +0x13

goroutine 35 [syscall]:
os/signal.signal_recv()
        /usr/local/go/src/runtime/sigqueue.go:152 +0x29
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:23 +0x13
created by os/signal.Notify.func1.1 in goroutine 1
        /usr/local/go/src/os/signal/signal.go:151 +0x1f

goroutine 6 [select]:
os/signal.NotifyContext.func1()
        /usr/local/go/src/os/signal/signal.go:288 +0x66
created by os/signal.NotifyContext in goroutine 1
        /usr/local/go/src/os/signal/signal.go:287 +0x165

goroutine 7 [IO wait]:
internal/poll.runtime_pollWait(0x7f43e9117f60, 0x72)
        /usr/local/go/src/runtime/netpoll.go:343 +0x85
internal/poll.(*pollDesc).wait(0xc000130180?, 0xc00038ded0?, 0x1)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc000130180, {0xc00038ded0, 0x10000, 0x10000})
        /usr/local/go/src/internal/poll/fd_unix.go:164 +0x27a
os.(*File).read(...)
        /usr/local/go/src/os/file_posix.go:29
os.(*File).Read(0xc0000b2008, {0xc00038ded0?, 0xe0bf50?, 0x1?})
        /usr/local/go/src/os/file.go:118 +0x52
github.com/fsnotify/fsnotify.(*Watcher).readEvents(0xc0001e7770)
        /home/alex/go/pkg/mod/github.com/fsnotify/fsnotify@v1.6.0/backend_inotify.go:356 +0xd2
created by github.com/fsnotify/fsnotify.NewWatcher in goroutine 1
        /home/alex/go/pkg/mod/github.com/fsnotify/fsnotify@v1.6.0/backend_inotify.go:150 +0x186

goroutine 8 [chan receive]:
fyne.io/fyne/v2/app.watchFile.func1()
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/app/settings_desktop.go:43 +0x66
created by fyne.io/fyne/v2/app.watchFile in goroutine 1
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/app/settings_desktop.go:42 +0xf8

goroutine 10 [chan receive]:
github.com/godbus/dbus/v5.newConn.func1()
        /home/alex/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/conn.go:303 +0x2c
created by github.com/godbus/dbus/v5.newConn in goroutine 1
        /home/alex/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/conn.go:302 +0x4f6

goroutine 50 [IO wait]:
internal/poll.runtime_pollWait(0x7f43e9117e68, 0x72)
        /usr/local/go/src/runtime/netpoll.go:343 +0x85
internal/poll.(*pollDesc).wait(0xc000164100?, 0xc000472760?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).ReadMsg(0xc000164100, {0xc000472760, 0x10, 0x10}, {0xc0001dd320, 0x1000, 0x1000}, 0xc00021bc04?)
        /usr/local/go/src/internal/poll/fd_unix.go:301 +0x38a
net.(*netFD).readMsg(0xc000164100, {0xc000472760?, 0xc00040a630?, 0x1523c98?}, {0xc0001dd320?, 0xc000472648?, 0xc00047264a?}, 0xc000472644?)
        /usr/local/go/src/net/fd_posix.go:78 +0x31
net.(*UnixConn).readMsg(0xc0000b2020, {0xc000472760?, 0xc00040a630?, 0x43f3eb?}, {0xc0001dd320?, 0x44af45?, 0xbb0400?})
        /usr/local/go/src/net/unixsock_posix.go:115 +0x45
net.(*UnixConn).ReadMsgUnix(0xc0000b2020, {0xc000472760?, 0xc00021bd90?, 0x0?}, {0xc0001dd320?, 0x436845?, 0xc000120d08?})
        /usr/local/go/src/net/unixsock.go:143 +0x36
github.com/godbus/dbus/v5.(*oobReader).Read(0xc0001dd300, {0xc000472760?, 0xc00021be00?, 0x436bc5?})
        /home/alex/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/transport_unix.go:21 +0x3c
io.ReadAtLeast({0xe104e0, 0xc0001dd300}, {0xc000472760, 0x10, 0x10}, 0x10)
        /usr/local/go/src/io/io.go:335 +0x90
io.ReadFull(...)
        /usr/local/go/src/io/io.go:354
github.com/godbus/dbus/v5.(*unixTransport).ReadMessage(0xc000120228)
        /home/alex/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/transport_unix.go:91 +0xfe
github.com/godbus/dbus/v5.(*Conn).inWorker(0xc000340f00)
        /home/alex/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/conn.go:390 +0x43
created by github.com/godbus/dbus/v5.(*Conn).Auth in goroutine 1
        /home/alex/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/auth.go:118 +0x945

goroutine 36 [chan receive]:
github.com/alexballas/go2tv/internal/gui.Start.func2()
        /home/alex/test/go2tv/internal/gui/gui.go:122 +0x25
created by github.com/alexballas/go2tv/internal/gui.Start in goroutine 1
        /home/alex/test/go2tv/internal/gui/gui.go:121 +0x4e7

goroutine 22 [IO wait]:
internal/poll.runtime_pollWait(0x7f43e9117d70, 0x72)
        /usr/local/go/src/runtime/netpoll.go:343 +0x85
internal/poll.(*pollDesc).wait(0xc00007e0c0?, 0xc0003bbed0?, 0x1)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc00007e0c0, {0xc0003bbed0, 0x10000, 0x10000})
        /usr/local/go/src/internal/poll/fd_unix.go:164 +0x27a
os.(*File).read(...)
        /usr/local/go/src/os/file_posix.go:29
os.(*File).Read(0xc000124060, {0xc0003bbed0?, 0x1083800010837?, 0x1083c0001083c?})
        /usr/local/go/src/os/file.go:118 +0x52
github.com/fsnotify/fsnotify.(*Watcher).readEvents(0xc000098280)
        /home/alex/go/pkg/mod/github.com/fsnotify/fsnotify@v1.6.0/backend_inotify.go:356 +0xd2
created by github.com/fsnotify/fsnotify.NewWatcher in goroutine 1
        /home/alex/go/pkg/mod/github.com/fsnotify/fsnotify@v1.6.0/backend_inotify.go:150 +0x186

goroutine 23 [chan receive]:
fyne.io/fyne/v2/app.watchFile.func1()
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/app/settings_desktop.go:43 +0x66
created by fyne.io/fyne/v2/app.watchFile in goroutine 1
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/app/settings_desktop.go:42 +0xf8

goroutine 24 [chan receive]:
fyne.io/fyne/v2/app.watchFreedekstopThemeChange()
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/app/app_xdg.go:194 +0x1c5
created by fyne.io/fyne/v2/app.watchTheme in goroutine 1
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/app/app_xdg.go:166 +0x1a

goroutine 26 [select]:
fyne.io/fyne/v2/internal/async.(*UnboundedFuncChan).processing(0xc0004d0c60)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/async/chan_func.go:45 +0xd5
created by fyne.io/fyne/v2/internal/async.NewUnboundedFuncChan in goroutine 1
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/async/chan_func.go:22 +0xe5

goroutine 27 [chan receive]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/common/window.go:34 +0x47
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/window.go:961 +0x139

goroutine 30 [select]:
fyne.io/fyne/v2/internal/async.(*UnboundedFuncChan).processing(0xc0013e6360)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/async/chan_func.go:45 +0xd5
created by fyne.io/fyne/v2/internal/async.NewUnboundedFuncChan in goroutine 1
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/async/chan_func.go:22 +0xe5

goroutine 31 [chan receive]:
fyne.io/fyne/v2/data/binding.queueItem.func1.1()
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/data/binding/queue.go:18 +0x4b
created by fyne.io/fyne/v2/data/binding.queueItem.func1 in goroutine 1
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/data/binding/queue.go:17 +0x46

goroutine 32 [IO wait]:
internal/poll.runtime_pollWait(0x7f43e9117a88, 0x72)
        /usr/local/go/src/runtime/netpoll.go:343 +0x85
internal/poll.(*pollDesc).wait(0xc00373a780?, 0xc002f7da90?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).RawRead(0xc00373a780, 0xc002f7da90)
        /usr/local/go/src/internal/poll/fd_unix.go:708 +0x12a
net.(*rawConn).Read(0xc00007c200, 0xc000068c00?)
        /usr/local/go/src/net/rawconn.go:44 +0x36
golang.org/x/net/internal/socket.(*Conn).recvMsg(0xc002b7a680, 0xc0031e9500, 0x0)
        /home/alex/go/pkg/mod/golang.org/x/net@v0.15.0/internal/socket/rawconn_msg.go:28 +0x152
golang.org/x/net/internal/socket.(*Conn).RecvMsg(...)
        /home/alex/go/pkg/mod/golang.org/x/net@v0.15.0/internal/socket/socket.go:247
golang.org/x/net/ipv4.(*payloadHandler).ReadFrom(0xc002d245b0, {0xc001ed6000, 0xffff, 0xffff})
        /home/alex/go/pkg/mod/golang.org/x/net@v0.15.0/ipv4/payload_cmsg.go:32 +0x229
github.com/alexballas/go-ssdp/internal/multicast.(*Conn).ReadPackets(0xc002b8d620, 0x77359400, 0xc0026fb8b8)
        /home/alex/go/pkg/mod/github.com/alexballas/go-ssdp@v0.0.3/internal/multicast/multicast.go:144 +0x9c
github.com/alexballas/go-ssdp.Search({0xccae9a, 0x8}, 0x2, {0x0, 0x0})
        /home/alex/go/pkg/mod/github.com/alexballas/go-ssdp@v0.0.3/search.go:107 +0x21b
github.com/alexballas/go2tv/devices.LoadSSDPservices(0x0?)
        /home/alex/test/go2tv/devices/devices.go:24 +0x98
github.com/alexballas/go2tv/internal/gui.getDevices(0xc003629520?)
        /home/alex/test/go2tv/internal/gui/actions.go:541 +0x25
github.com/alexballas/go2tv/internal/gui.refreshDevList(0xc0004ea1e0, 0xc000012168)
        /home/alex/test/go2tv/internal/gui/main.go:528 +0x112
created by github.com/alexballas/go2tv/internal/gui.mainWindow in goroutine 1
        /home/alex/test/go2tv/internal/gui/main.go:507 +0x2358

goroutine 33 [chan receive]:
github.com/alexballas/go2tv/internal/gui.checkMutefunc(0xc0004ea1e0)
        /home/alex/test/go2tv/internal/gui/main.go:587 +0x59
created by github.com/alexballas/go2tv/internal/gui.mainWindow in goroutine 1
        /home/alex/test/go2tv/internal/gui/main.go:511 +0x239b

goroutine 66 [chan receive]:
github.com/alexballas/go2tv/internal/gui.sliderUpdate(0xc0004ea1e0)
        /home/alex/test/go2tv/internal/gui/main.go:622 +0x4d
created by github.com/alexballas/go2tv/internal/gui.mainWindow in goroutine 1
        /home/alex/test/go2tv/internal/gui/main.go:515 +0x23dd

goroutine 38 [chan receive]:
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).catchTerm(0x0?)
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/driver_desktop.go:172 +0x85
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run in goroutine 1
        /home/alex/go/pkg/mod/fyne.io/fyne/v2@v2.4.1-0.20231003212356-ac320479eb47/internal/driver/glfw/driver.go:167 +0x65
Makefile:22: recipe for target 'run' failed
make: *** [run] Error 2
```

--- a/dialog/fileitem.go
+++ b/dialog/fileitem.go
@@ -102,7 +102,6 @@ func (s *fileItemRenderer) Layout(size fyne.Size) {
 		s.text.Resize(fyne.NewSize(size.Width, textMin.Height))
 		s.text.Move(fyne.NewPos(fileInlineIconSize, (size.Height-textMin.Height)/2))
 	}
-	s.text.Refresh()
 }
 
 func (s *fileItemRenderer) MinSize() fyne.Size {

